python/sglang/srt/managers/tokenizer_manager.py

```diff
@@ -1190,7 +1190,9 @@ def abort_request(self, rid: str = "", abort_all: bool = False):
     async def pause_generation(self):
         async with self.is_pause_cond:
             self.is_pause = True
-            self.abort_request(abort_all=True)
+            while self.model_update_lock.is_locked():
+                self.abort_request(abort_all=True)
```

Comment:
(optional) btw is it possible we await this (using Communicator mechanism), instead of fire-and-forget

Reply:
yeah, this really makes sense, and the barrier now is that `abort_request` is called as a non-async function in many locations. we may need to have another async version of abort request, which may be a non-trivial refactor.

Reply:
yes let's skip it now and may be add a todo comment instead